### DISCUSSION TITLE
Rediscovering DevOps

### DISCUSSION THREAD

**Discussion by relaxdiego** [relaxdiego/relaxdiego.github.com#31] [General]
This is the thread for discussing https://relaxdiego.com/2021/05/rediscovering-devops.html

**Comment by relaxdiego** [comment]
Link to a LinkedIn comment thread discussing the above blog post:

https://www.linkedin.com/feed/update/urn:li:activity:6993192173496143872?commentUrn=urn%3Ali%3Acomment%3A%28activity%3A6993192173496143872%2C6994071084002086912%29

**Comment by markpanizales** [comment]
Hi Mark,

First of all thanks for sharing this article that talks about DevOps. Like you, I am also passionate about providing digital transformation to organizations that would result in a high-performance team.

What I like about your blog post is that it is very much aligned with our goal of transforming teams or squads into better versions of themselves through communication and collaboration. The [Phoenix Project](https://www.amazon.com/Phoenix-Project-DevOps-Helping-Business/dp/0988262592) set a lot of the theory on how operations and developers can work together.

My journey with DevOps was indeed a roller coaster ride and I would like to share some insights that I always strive to achieve:

1. **Freedom and Responsibility**
- Traditional policies and procedures prevent people from accessing the production server.
	- Push windows and schedules during business hours
	- A crusade process that people need to go through in order to have their code blessed for production deployment
- Focus on having federal governance that allows the squad to be isolated and own the process

2. **Continuous Innovation**
- Automate toil so developers can focus on features, and new things to delight our customers
- Create an environment that will keep our engineers doing things that are fun, interesting, and exciting things for those customers

3. **Process and Procedures**
- Bureaucratic
	- Lots of centralized guard rails and misunderstandings of anti-patterns 
	- Lots of processes and procedures in order to get the output of bureaucracy as the standard ways of working
- Trust
	- Traditional leaders are obsessed with control to the point that we build a bureaucratic machine
	- Big individuals are dreamers and passionate to try new things and ideas, and have a voice to challenge the traditional system
	- People prefer context over control
		- Involve squad leaders to have visibility in the context of the roadmap to have better insights and approach in providing a solution
- Enablement
	- Focus on having people that are software engineers rather than enablers
	- Make ownership easy: you build it, you run it.
	- Invest in data, algorithms, and systems that can groom all the data very quickly and let us know when there is a problem

**Summary**
It is more of the mindset, culture, and set of technical best practices which created the wonderful results of DevOps
